### DISCUSSION TITLE
如何让公式里面的数字字符与文本中数字字符统一？

### DISCUSSION THREAD

**Discussion by earthmoon** [whutug/whu-thesis#157] [Q&A] [answered]
你好，

```
数量$\leq 55$个，至少有50个
```
这段代码生成的数字5不一致，`50 `中的5是文本的5，`55`中的5是数学形式的5，很多这样的情况让排版看起来混乱：
![image](https://user-images.githubusercontent.com/45897471/173060686-ea901422-7e9b-41f4-a09f-ba9ff4fcdc2c.png)
该怎么做让全文的数字能够统一？
谢谢

**Comment by xkwxdyy** [comment]
你自己不是知道问题所在吗  把所有出现的数字放在 `$...$` 中，没有什么投机取巧的办法

**Reply by earthmoon** [reply]
表格中大量数字都这么做，工作量就太大了。而且比如我们引用表格`表 1.1`，这里的`1.1` 无法通过`$...$`改过来字符。所以有没有办法让`$...$`中的数字采用文本数字，但是`$...$`中符号保持不动？

**Reply by xkwxdyy** [reply]
那你就所有数字都不要。
其次，这个问题是你自己现在才发现的，工作量也是你自己造成的。

**Reply by earthmoon** [reply]
> 那你就所有数字都不要。 其次，这个问题是你自己现在才发现的，工作量也是你自己造成的。

我再想想办法吧，即使没有什么工作量，我也会去寻找在字体设置上修改过来而不是逐个解决。谢谢了

**Comment by sikouhjw** [comment]
统一用 `XITS` 和 `XITS Math` 不就得了？

**Reply by earthmoon** [reply]
可以详细说说吗，第一次看到`XITS `和 `XITS Math`

**Reply by sikouhjw** [reply]
> 可以详细说说吗，第一次看到`XITS `和 `XITS Math`

多看手册
https://github.com/whutug/whu-thesis/blob/6f19c2e0d34bbfc752f86a7ea505154a185ff595/whu-thesis-doc.tex#L364-L406

**Reply by earthmoon** [reply]
这该怎么改呀

**Reply by xkwxdyy** [reply]
> > 可以详细说说吗，第一次看到`XITS `和 `XITS Math`
> 
> 多看手册
> 
> https://github.com/whutug/whu-thesis/blob/6f19c2e0d34bbfc752f86a7ea505154a185ff595/whu-thesis-doc.tex#L364-L406

不过数学字体我在 dev 解决 `mathcal` 和 `mathscr` 问题的时候改了一下
https://github.com/whutug/whu-thesis/commit/8e171c004fad890d7ff9237b3a3f262abb6693cc#diff-7d6f3ec874492c997f3562613b1ba34c85df8dd64a3e0fd69ceb2da323f7a070R637-R642

**Reply by earthmoon** [reply]
谢谢~这样做了后，感觉表格里面的数字和文本数字不一样哎。我的表格用`tabularray`制作的

**Reply by sikouhjw** [reply]
> 谢谢~这样做了后，感觉表格里面的数字和文本数字不一样哎。我的表格用`tabularray`制作的

哪里不一样？截图都不给？

**Comment by earthmoon** [comment]
目前的一个解决方案->让全文所有数字都转为`$..$`中的数字字体：在`whu-thesis-demo.tex`中
```
style              =
      {
        graphics-path  = {{figures/}{data/}},
```
改为：

```
style              =
      {
      	 font=termes, math-font=termes,
        graphics-path  = {{figures/}{data/}},
```

**Comment by earthmoon** [comment, marked as answer]
如果需要把公式里面的数字字体转为文本中的数字字体，在`whu-thesis-demo.dex`文档中，找到`\begin{document}`，在这一行下面添加`\setmathfont[range=up/{num}]{Times New Roman}`就可以了

**Reply by sikouhjw** [reply]
XITS 是 Times New Roman 的复刻。我不明白都设置 XITS 怎么还会不同。

**Reply by earthmoon** [reply]
你好呀，我刚试了下，有一点点区别：
`55\par \num{55}\par $55$`
效果如下：
![image](https://user-images.githubusercontent.com/45897471/173178400-1aeed533-769c-4191-9ec6-5ea339839afb.png)
是不是有点不一样，然后设置代码如下：
![image](https://user-images.githubusercontent.com/45897471/173178413-15f3ee9e-b9b2-4c1d-a298-d598f44e906d.png)

**Reply by earthmoon** [reply]
`tabularray`中的数字是采用文本数字，所以和公式中的数字有点不一样。

**Reply by tanukihee** [reply]
> XITS 是 Times New Roman 的复刻。我不明白都设置 XITS 怎么还会不同。

这个可以理解，毕竟克隆也不可能完全一样。但是在 12pt 这个尺度下，在乎这种不一样是完全没有意义的

**Comment by tanukihee** [comment]
在 12pt 这个尺度下，在乎这种不一样是完全没有意义的

**Reply by earthmoon** [reply]
看的出来的，当一列数据是文本数字，一列是数学数字就特别明显，这样就出现了表格中两种数字字体，我当时就是在这个情况下才决定一定要统一数学字符

> \begin{table}
> \begin{tblr}{}
> 55 &\SI{55}{\%}\\\\
> \end{tblr}
> \end{table}

![image](https://user-images.githubusercontent.com/45897471/173224234-595856dd-0ecb-477e-bdbe-3c3883dc58a8.png)
